### PR TITLE
Docker and form run

### PR DESCRIPTION
**Pull request by BirdOnTheBranch**
Correcting all conflicts, The app works correctly

### CODE REVIEW

**Comment by thecodix** [comment]
Please solve conflicts

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,112 @@
+
```
Comment (thecodix) [inline .gitignore:1]: There is a lot of stuff here. Let's stick to the things we need

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM python:3.8-slim-buster
```
Comment (thecodix) [inline Dockerfile:1]: Check this comment about optimisation of requirements file
https://stackoverflow.com/a/34399661

Also there is many blankspaces

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        build: .
+        ports:
+            - "5000:5000"
+        volumes:
+            - .:/psicoquestions
+        depends_on:
+            - redis
+    redis:
+        image: redis
+
```
Comment (thecodix) [inline docker-compose.yml:14]: Many blankspaces in the bottom

File: questions/main/controllers.py [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
 
 
-@main_bp.route('/all')
+@main_bp.route('/all', methods = ['GET','POST'])
 def display_questions():
     questions = utils.get_questions()
-    return render_template("questions.htm", questions=questions)
+    form = forms.QuestionForm(request.form)
+    if request.method == 'POST':
+        print(form.answer.data)
```
Comment (thecodix) [inline questions/main/controllers.py:17]: If there is only a print here, we don't need the 'if'

File: questions/main/controllers.py [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
+
+    return render_template("questions.htm", questions=questions, form=form)
```
Comment (thecodix) [inline questions/main/controllers.py:19]: Add a blank line in the bottom

File: questions/templates/layout.htm [hunk 6]
```diff
@@ -3,6 +3,10 @@
   <head>
     <title>Psychology Questions</title>
     <meta charset="utf-8">
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
+    <script src="https://code.jquery.com/jquery-3.4.1.slim.min.js" integrity="sha384-J6qa4849blE2+poT4WnyKhv5vZF5SrPo0iEjwBvKU7imGFAV0wwj1yYfoRSJoZ+n" crossorigin="anonymous"></script>
```
Comment (thecodix) [inline questions/templates/layout.htm:7]: Can you check if we can load js libraries at the bottom of the page?
Also, do we need them?

File: run.py [hunk 7]
```diff
@@ -2,4 +2,4 @@
 
 if __name__ == '__main__':
     app = create_app()
-    app.run()
+    app.run(host='0.0.0.0', debug='True')
```
Comment (thecodix) [inline run.py:5]: We shouldn't add this as default configuration

File: .gitignore (hunk 1 above)
Comment (BirdOnTheBranch) [reply .gitignore:1]: Okay. It is a snipet that I use by default for python. Now I have taken away some things.

File: Dockerfile (hunk 2 above)
Comment (BirdOnTheBranch) [reply Dockerfile:1]: Im changed it.

File: docker-compose.yml (hunk 3 above)
Comment (BirdOnTheBranch) [reply docker-compose.yml:14]: Really. Only one white space in the last line to the code, is a good practice.

File: questions/main/controllers.py (hunk 4 above)
Comment (BirdOnTheBranch) [reply questions/main/controllers.py:17]: I will prepare for the future but not necesary, you're right. I change it right now.

File: questions/main/controllers.py (hunk 5 above)
Comment (BirdOnTheBranch) [reply questions/main/controllers.py:19]: Ok.

File: questions/templates/layout.htm (hunk 6 above)
Comment (BirdOnTheBranch) [reply questions/templates/layout.htm:7]: Is not necessary. I changed it.

File: run.py (hunk 7 above)
Comment (BirdOnTheBranch) [reply run.py:5]: The host parameter its necessary for correct work of Docker.I have only deleted the "debug" parameter.

File: .gitignore (hunk 1 above)
Comment (thecodix) [reply .gitignore:1]: It is still huge, I'd say nothing after line 35 really applies here

File: questions/main/controllers.py [hunk 8]
```diff
@@ -1,14 +1,19 @@
...
     return render_template("index.htm")
 
 
-@main_bp.route('/all')
+@main_bp.route('/all', methods = ['GET','POST'])
 def display_questions():
     questions = utils.get_questions()
-    return render_template("questions.htm", questions=questions)
+    form = forms.QuestionForm(request.form)
+    print(form.answer.data)
```
Comment (thecodix) [inline questions/main/controllers.py:16]: print is only for debugging, so don't include it on commits to be merged on master

File: questions/templates/layout.htm [hunk 9]
```diff
@@ -3,6 +3,8 @@
   <head>
     <title>Psychology Questions</title>
     <meta charset="utf-8">
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css" integrity="sha384-Vkoo8x4CGsO3+Hhxv8T/Q5PaXtkKtu6ug5TOeNV6gBiFeWPGFN9MuhOf23Q9Ifjh" crossorigin="anonymous">
+
```
Comment (thecodix) [inline questions/templates/layout.htm:7]: remove blankspace

File: questions/templates/questions.htm [hunk 10]
```diff
@@ -1,11 +1,15 @@
 {% extends 'layout.htm' %}
+
 {% block container %}
-  {% for question in questions %}
+  
+{% for question in questions %}
```
Comment (thecodix) [inline questions/templates/questions.htm:5]: The indent here is not consistent with the endfor indent

File: questions/main/controllers.py (hunk 8 above)
Comment (BirdOnTheBranch) [reply questions/main/controllers.py:16]: I solved.

File: questions/templates/layout.htm (hunk 9 above)
Comment (BirdOnTheBranch) [reply questions/templates/layout.htm:7]: Ok.

File: questions/templates/questions.htm (hunk 10 above)
Comment (BirdOnTheBranch) [reply questions/templates/questions.htm:5]: I deleted the indent.

File: .gitignore (hunk 1 above)
Comment (BirdOnTheBranch) [reply .gitignore]: I have shortened it more.

**Comment by thecodix** [comment]
Can you update the README.md with the commands to make the app run and how to install it?

**Pull request merged**